core/src/main/java/org/testcontainers/containers/wait/internal/InternalCommandPortListeningCheck.java

```diff
@@ -24,22 +23,17 @@
 
     @Override
     public Boolean call() {
-        StringBuilder command = new StringBuilder("true");
-
-        for (int internalPort : internalPorts) {
-            command.append(" && ");
-            command.append(" (");
-            command.append(format("cat /proc/net/tcp* | awk '{print $2}' | grep -i ':0*%x'", internalPort));
-            command.append(" || ");
-            command.append(format("nc -vz -w 1 localhost %d", internalPort));
-            command.append(" || ");
-            command.append(format("/bin/bash -c '</dev/tcp/localhost/%d'", internalPort));
-            command.append(")");
-        }
+         String command = internalPorts.stream()
+             .map(it -> String.format("(cat /proc/net/tcp* | awk '{print $2}' | grep -i ':0*%x')", it))
```

Comment:
After some considerations, I came to a conclusion that `cat` + `/proc/net/tcp*` should work with any shell, so we don't need `nc` and `bash`-based strategies, as they would only add an overhead